### PR TITLE
Feature: Enable using logged in account for broker

### PR DESCRIPTION
**Pull request by billwert**
This adds a new method to `InteractiveBrowserBrokerCredentialBuilder` to enable using the logged in account for WAM broker scenarios. By default the broker UI will be shown.

### CODE REVIEW

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java [hunk 1]
```diff
@@ -781,6 +782,16 @@ public IdentityClientOptions setEnableLegacyMsaPassthrough(boolean enableMsaPass
         return this;
     }
 
+    /**
+     * Sets whether to use the logged-in user's account for broker authentication.
+     * @param useOperatingSystemAccount
+     * @return the updated client options
+     */
+    public IdentityClientOptions setUseOperatingSystemAccount(boolean useOperatingSystemAccount) {
```
Comment (joshfree) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java:790]: @christothes is this the name we've picked for all languages

Comment (joshfree) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java:790]: Answered my own question --- https://github.com/Azure/azure-sdk-for-net/pull/39607/files

**Review by joshfree — APPROVED** [review]

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java (hunk 1 above)
Comment (billwert) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClientOptions.java:790]: worth noting this is the internal property bag. The public API is `useOperatingSystemAccount` on the [builder here](https://github.com/Azure/azure-sdk-for-java/pull/38792/files#diff-6bf12785e338eb151b9328776f5ee2dd11623f7028c0873edc1910105b58bbebR70).

File: sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java [hunk 2]
```diff
@@ -216,53 +216,37 @@ public AccessToken authenticateWithConfidentialClientCache(TokenRequestContext r
...
-        }
-
-        SilentParameters.SilentParametersBuilder forceParametersBuilder = SilentParameters.builder(
-            new HashSet<>(request.getScopes())).forceRefresh(true);
-
         if (request.isCaeEnabled() && request.getClaims() != null) {
             ClaimsRequest customClaimRequest = CustomClaimRequest
                 .formatAsClaimsRequest(request.getClaims());
-            forceParametersBuilder.claims(customClaimRequest);
+            parametersBuilder.claims(customClaimRequest);
```
Comment (g2vinay) [inline sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java:240]: forceRefresh should set to true here ?

Comment (billwert) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java:240]: It is set true in the conditional on line 233.

Comment (g2vinay) [reply sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java:240]: The equivalent async flow is setting it to true here:

https://github.com/Azure/azure-sdk-for-java/pull/38792/files/dfb75b8dad25e94487f25c5e93972a906085900f#diff-0983e9433b1bbd62e50b6b0546ed7f9d694351e56249a8af0683b9eafa208713R638

**Pull request merged**